src/components/Testssl/Testssl.tsx

```diff
@@ -0,0 +1,288 @@
+        } else {
+            setOutput((prevOutput) => prevOutput + `\nScan terminated with exit code: ${code}`);
+        }
+        setLoading(false);
+        setAllowSave(true);
+        setHasSaved(false);
+    }, []);
+
+    /**
+     * Handle form submission by constructing and executing the testssl command.
+     * @param values - The form values containing the scan configuration.
+     */
+    const onSubmit = async (values: FormValuesType) => {
+        setLoading(true);
+        setAllowSave(false);
+        setOutput("");
+
+        const args: string[] = [];
+
+        // Add command-line arguments based on form values
+        if (values.testProtocols) args.push("-p");
+        if (values.testHttpHeaders) args.push("-h");
+        if (values.testServerPreferences) args.push("-s");
+        if (values.testForwardSecrecy) args.push("-f");
+        if (values.testVulnerabilities) args.push("-U");
+        values.severityLevel.forEach((level) => {
+            args.push("--severity");
+            args.push(level);
+        });
+        if (values.sneakyMode) args.push("--sneaky");
+        if (values.idsFriendly) args.push("--ids-friendly");
+
+        // Add the target at the end of the argument list
+        args.push(values.target);
+
+        try {
+            const result = await CommandHelper.runCommand("testssl", args);
+            handleProcessData(result);
+            handleProcessTermination(0);
+        } catch (error: unknown) {
+            const errorMessage = error instanceof Error ? error.message : "An unknown error occurred";
+            handleProcessData(errorMessage);
+            handleProcessTermination(1);
+        }
+    };
+
+    /**
+     * Clear the output and reset related states.
+     * This function resets the output to an empty string and updates the save-related states.
+     */
+    const clearOutput = useCallback(() => {
+        setOutput("");
+        setHasSaved(false);
+        setAllowSave(false);
+    }, []);
+
+    /**
+     * Handle the completion of saving the output.
+     * This function updates the state to reflect that the output has been saved
+     * and prevents further saves until new output is generated.
+     */
+    const handleSaveComplete = () => {
+        setHasSaved(true);
+        setAllowSave(false);
+    };
+
+    return (
+        <RenderComponent
+            title={title}
+            description={description}
+            steps={steps}
+            tutorial={tutorial}
+            sourceLink={sourceLink}
+        >
+            {!loadingModal && (
+                <InstallationModal
+                    isOpen={opened}
+                    setOpened={setOpened}
+                    feature_description={description}
+                    dependencies={dependencies}
+                ></InstallationModal>
+            )}
+            <form onSubmit={form.onSubmit(onSubmit)}>
+                {LoadingOverlayAndCancelButton(loading, pid)}
```

Comment:
This change had an unintended consequence. Since you are using CommandHelper.runCommand rather than runCommandGetPidAndOutput, you do not actually use a pid variable. Is there a reason why you are using CommandHelper.runCommand rather than runCommandGetPidAndOutPut? 

If you must use CommandHelper.runCommand, then you may need to remove this line and go without an overlay.